src/realmd/AuthSocket.cpp

```diff
@@ -198,51 +179,66 @@ AuthSocket::~AuthSocket()
         ACE_OS::close(patch_);
 }
 
-/// Accept the connection and set the s random value for SRP6
-void AuthSocket::OnAccept()
-{
-    BASIC_LOG("Accepting connection from '%s'", get_remote_address().c_str());
-}
-
 /// Read the packet from the client
-void AuthSocket::OnRead()
+bool AuthSocket::ProcessIncomingData()
 {
-    uint8 _cmd;
-    while (1)
+    // benchmarking has demonstrated that this lookup method is faster than std::map
```

Comment:
Just out of interest, is this lookup method also faster than std::unordered_map?
Not that it matters because for an array this small.